backend/canisters/community/impl/src/lib.rs

```diff
@@ -749,23 +750,44 @@ impl Data {
         }
     }
 
-    pub fn get_member_for_events(&self, caller: Principal) -> Result<Option<CommunityMemberInternal>, EventsResponse> {
-        let hidden_for_non_members = !self.is_public.value || self.has_payment_gate();
-        let member = self.members.get(caller);
-
-        if hidden_for_non_members {
-            if let Some(member) = &member {
-                if member.suspended().value {
-                    return Err(EventsResponse::UserSuspended);
-                } else if member.lapsed().value {
-                    return Err(EventsResponse::UserLapsed);
-                }
-            } else {
+    pub fn get_caller_for_events(
+        &self,
+        caller: Principal,
+        channel_id: ChannelId,
+        bot_initiator: Option<BotInitiator>,
+    ) -> Result<EventsCaller, EventsResponse> {
+        if let Some(initiator) = bot_initiator {
+            let bot_user_id = caller.into();
+            let Some(permissions) = self.granted_bot_permissions(&bot_user_id, &initiator, Some(channel_id)) else {
+                return Err(EventsResponse::UserNotInCommunity);
```

Comment:
This is a bit iffy. I just reused the existing `UserNotInCommunity` response rather than introduce a new one.
They'll all disappear once we introduce service wide error codes (which we'll hopefully do soon) anyway.